st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -277,17 +294,35 @@ def validate(self):
 
             return self
 
-    @jsexpose(body_cls=ExecutionSpecificationAPI, status_code=http_client.CREATED)
-    def post(self, spec_api, execution_id, no_merge=False):
+    def post(self, spec_api, id, requester_user=None, no_merge=False, **kwargs):
         """
         Re-run the provided action execution optionally specifying override parameters.
 
         Handles requests:
 
             POST /executions/<id>/re_run
         """
+
+        if (spec_api.tasks or spec_api.reset) and spec_api.parameters:
```

Comment:
The problem with that is that API is not the only way for us to get data into the system.

Some resources are also registered using `st2-register-content` script, etc. and having some centralized place (in this case a model) where a common validation happens, etc. is better - this way we can be sure it's consistently applied everywhere where it needs to be.

Having said that, yes, this is not really an issue for `re_run` itself so I'm fine with it being the way it is, but it is when "creating" other resources.